### ISSUE TITLE
[BUG] multiple tests related to `FRESH` fail on all platforms and python versions

### ISSUE THREAD

**Issue by lmmentel** [alan-turing-institute/sktime#1803]
**Describe the bug**

I found another test failing on all platforms and python versions.

<details>
<summary>test_fresh_prince_on_unit_test_data</summary>

```python
================================== FAILURES ===================================
_____________________ test_fresh_prince_on_unit_test_data _____________________
[gw0] win32 -- Python 3.8.5 C:\Miniconda\envs\test\python.exe

    def test_fresh_prince_on_unit_test_data():
        """Test of FreshPRINCE on unit test data."""
        # load unit test data
        X_train, y_train = load_unit_test(split="train")
        X_test, y_test = load_unit_test(split="test")
        indices = np.random.RandomState(0).choice(len(y_train), 10, replace=False)
    
        # train FreshPRINCE classifier
        fp = FreshPRINCE(
            random_state=0,
            default_fc_parameters="minimal",
            n_estimators=10,
            save_transformed_data=True,
        )
        fp.fit(X_train, y_train)
    
        # assert probabilities are the same
        probas = fp.predict_proba(X_test.iloc[indices])
>       testing.assert_array_almost_equal(probas, fp_classifier_unit_test_probas, decimal=2)
E       AssertionError: 
E       Arrays are not almost equal to 2 decimals
E       
E       Mismatched elements: 10 / 20 (50%)
E       Max absolute difference: 0.2
E       Max relative difference: 1.
E        x: array([[0.2, 0.8],
E              [1. , 0. ],
E              [0. , 1. ],...
E        y: array([[0.2, 0.8],
E              [1. , 0. ],
E              [0.1, 0.9],...

X_test     =                                                 dim_0
0   0      501.0
1      328.0
2      195.0
3      ...
1   0     ...
3      ...
20  0       95.0
1       63.0
2       45.0
3      ...
21  0      193.0
1      195.0
2       90.0
3      ...
X_train    =                                                 dim_0
0   0      573.0
1      375.0
2      301.0
3      ...
1   0     ...
3      ...
18  0       95.0
1       63.0
2       45.0
3      ...
19  0       82.0
1       93.0
2       51.0
3      ...
fp         = FreshPRINCE(chunksize=None, default_fc_parameters='minimal', n_estimators=10,
            n_jobs=1, random_state=0, save_transformed_data=True, verbose=0)
indices    = array([18,  1, 19,  8, 10, 17,  6, 13,  4,  2])
probas     = array([[0.2, 0.8],
       [1. , 0. ],
       [0. , 1. ],
       [1. , 0. ],
       [1. , 0. ],
       [1. , 0. ],
       [1. , 0. ],
       [1. , 0. ],
       [1. , 0. ],
       [1. , 0. ]])
y_test     = array(['1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '2',
       '2', '2', '2', '2', '2', '2', '2', '2', '2'], dtype='<U1')
y_train    = <details>
<summary>test_fresh_prince_on_unit_test_data</summary>array(['1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '2', '2', '2',
       '2', '2', '2', '2', '2', '2', '2'], dtype='<U1')

C:\Miniconda\envs\test\lib\site-packages\sktime\classification\feature_based\tests\test_fresh_prince.py:29: AssertionError
```

</details>

<details>
<summary>test_tsfresh_classifier_on_unit_test_data failure</summary>

```python
__________________ test_tsfresh_classifier_on_unit_test_data __________________
[gw0] win32 -- Python 3.8.5 C:\Miniconda\envs\test\python.exe

    def test_tsfresh_classifier_on_unit_test_data():
        """Test of TSFreshClassifier on unit test data."""
        # load unit test data
        X_train, y_train = load_unit_test(split="train")
        X_test, y_test = load_unit_test(split="test")
        indices = np.random.RandomState(0).choice(len(y_train), 10, replace=False)
    
        # train TSFresh classifier
        tsfc = TSFreshClassifier(
            random_state=0,
            default_fc_parameters="minimal",
            relevant_feature_extractor=False,
            estimator=RandomForestClassifier(n_estimators=10),
        )
        tsfc.fit(X_train, y_train)
    
        # assert probabilities are the same
        probas = tsfc.predict_proba(X_test.iloc[indices])
>       testing.assert_array_almost_equal(
            probas, tsfresh_classifier_unit_test_probas, decimal=2
        )
E       AssertionError: 
E       Arrays are not almost equal to 2 decimals
E       
E       Mismatched elements: 10 / 20 (50%)
E       Max absolute difference: 0.2
E       Max relative difference: 2.
E        x: array([[0.1, 0.9],
E              [0.7, 0.3],
E              [0. , 1. ],...
E        y: array([[0. , 1. ],
E              [0.9, 0.1],
E              [0. , 1. ],...

X_test     =                                                 dim_0
0   0      501.0
1      328.0
2      195.0
3      ...
1   0     ...
3      ...
20  0       95.0
1       63.0
2       45.0
3      ...
21  0      193.0
1      195.0
2       90.0
3      ...
X_train    =                                                 dim_0
0   0      573.0
1      375.0
2      301.0
3      ...
1   0     ...
3      ...
18  0       95.0
1       63.0
2       45.0
3      ...
19  0       82.0
1       93.0
2       51.0
3      ...
indices    = array([18,  1, 19,  8, 10, 17,  6, 13,  4,  2])
probas     = array([[0.1, 0.9],
       [0.7, 0.3],
       [0. , 1. ],
       [0.9, 0.1],
       [0.8, 0.2],
       [1. , 0. ],
       [0.6, 0.4],
       [0.9, 0.1],
       [1. , 0. ],
       [0.9, 0.1]])
tsfc       = TSFreshClassifier(chunksize=None, default_fc_parameters='minimal',
                  estimator=RandomForestClassifier(...tart=False),
                  n_jobs=1, random_state=0, relevant_feature_extractor=False,
                  verbose=0)
y_test     = array(['1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '2',
       '2', '2', '2', '2', '2', '2', '2', '2', '2'], dtype='<U1')
y_train    = array(['1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '2', '2', '2',
       '2', '2', '2', '2', '2', '2', '2'], dtype='<U1')

C:\Miniconda\envs\test\lib\site-packages\sktime\classification\feature_based\tests\test_tsfresh_classifier.py:29: AssertionError
```

</details>

<details>
<summary>test_tsfresh_classifier_on_basic_motions</summary>

```python
__________________ test_tsfresh_classifier_on_basic_motions ___________________
[gw1] win32 -- Python 3.8.5 C:\Miniconda\envs\test\python.exe

    def test_tsfresh_classifier_on_basic_motions():
        """Test of TSFreshClassifier on basic motions."""
        # load basic motions data
        X_train, y_train = load_basic_motions(split="train")
        X_test, y_test = load_basic_motions(split="test")
        indices = np.random.RandomState(4).choice(len(y_train), 10, replace=False)
    
        # train TSFresh classifier
        tsfc = TSFreshClassifier(
            random_state=0,
            default_fc_parameters="minimal",
            relevant_feature_extractor=False,
            estimator=RandomForestClassifier(n_estimators=10),
        )
        tsfc.fit(X_train.iloc[indices], y_train[indices])
    
        # assert probabilities are the same
        probas = tsfc.predict_proba(X_test.iloc[indices])
>       testing.assert_array_almost_equal(
            probas, tsfresh_classifier_basic_motions_probas, decimal=2
        )
E       AssertionError: 
E       Arrays are not almost equal to 2 decimals
E       
E       Mismatched elements: 14 / 40 (35%)
E       Max absolute difference: 0.2
E       Max relative difference: 1.
E        x: array([[0. , 0. , 0.2, 0.8],
E              [0.4, 0.3, 0. , 0.3],
E              [0. , 0. , 0.8, 0.2],...
E        y: array([[0. , 0. , 0.2, 0.8],
E              [0.4, 0.2, 0.1, 0.3],
E              [0. , 0. , 0.9, 0.1],...

X_test     =                                                 dim_0  ...                                              dim_5
0   0   ...901645
1      0.901645
2     -0.05469...  ...  0     0.245030
1     0.245030
2     0.005327
3...

[40 rows x 6 columns]
X_train    =                                                 dim_0  ...                                              dim_5
0   0   ...11973
1     1.211973
2    -0.605948
3...  ...  0    -0.042614
1    -0.042614
2    -0.095881
3...

[40 rows x 6 columns]
indices    = array([28, 39,  4, 15, 26, 20, 31,  7, 16, 11])
probas     = array([[0. , 0. , 0.2, 0.8],
       [0.4, 0.3, 0. , 0.3],
       [0. , 0. , 0.8, 0.2],
       [0. , 1. , 0. , 0. ],
  ...],
       [0.5, 0.3, 0. , 0.2],
       [0. , 0. , 0.9, 0.1],
       [0.1, 0.9, 0. , 0. ],
       [0. , 1. , 0. , 0. ]])
tsfc       = TSFreshClassifier(chunksize=None, default_fc_parameters='minimal',
                  estimator=RandomForestClassifier(...tart=False),
                  n_jobs=1, random_state=0, relevant_feature_extractor=False,
                  verbose=0)
y_test     = array(['standing', 'standing', 'standing', 'standing', 'standing',
       'standing', 'standing', 'standing', 'standin...
       'badminton', 'badminton', 'badminton', 'badminton', 'badminton',
       'badminton', 'badminton'], dtype='<U9')
y_train    = array(['standing', 'standing', 'standing', 'standing', 'standing',
       'standing', 'standing', 'standing', 'standin...
       'badminton', 'badminton', 'badminton', 'badminton', 'badminton',
       'badminton', 'badminton'], dtype='<U9')

C:\Miniconda\envs\test\lib\site-packages\sktime\classification\feature_based\tests\test_tsfresh_classifier.py:52: AssertionError
```

</details>

**To Reproduce**

In order to reproduce you need to bump statsmodels to `>=0.13` or use sktime version from #1620 PR.

To recreate the env you can use:

```bash
conda install -c anaconda libpython
conda install -c anaconda "pystan==2.19.1.1"
conda install -c conda-forge "fbprophet>=0.7.1"
conda install -c conda-forge scipy cython matplotlib
python -m pip install .[all_extras,dev]
```

then run the test with:

```bash
python -m pytest --showlocals --durations=10 --cov-report=xml -n 2 --cov=sktime -v --pyargs sktime
```

**Expected behavior**

Test passes
**Additional context**

**Versions**
<details>

</details>

**Comment by lmmentel** [comment]
Might be related to this message (extracted from CI):

```bash
tsfresh 0.19.0 requires statsmodels>=0.13, but you'll have statsmodels 0.12.1 which is incompatible.
```

**Comment by lmmentel** [comment]
After bumping `statsmodels` to `>=0.13` I now have the same tests failing on all platforms. It seems that by fixing a dependency conflict I found some more failing tests.  @MatthewMiddlehurst I saw your name in the `sktime/classification/feature_based/_tsfresh_classifier.py` module, would you be able to help with this?

**Comment by MatthewMiddlehurst** [comment]
These tests compare the current outputs from the classifier to a set of saved results. Is there any good reason why this would change the results? If so, we can just update the reference results.

I assume the TSFresh transform results are different as well? How much do they change from the previous version?

**Comment by lmmentel** [comment]
Thanks for the input @MatthewMiddlehurst. 

> Is there any good reason why this would change the results?

The thing that I can say for sure is that there was a dependency conflict since tsfresh has `statsmodels>=0.13` however sktime has essentially `statsmodels==0.12.1` which is a conflict that pip is apparently happy to ignore. Once I bumped `statsmodels>=0.13` the fresh related tests failed. I haven't investigated further at this point.

> I assume the TSFresh transform results are different as well? How much do they change from the previous version?

That is possible although I haven't seen any other fresh related tests failing. 

Let me know if you have the appetite to tackle this, otherwise I will simply update the expected values for tests.

**Comment by MatthewMiddlehurst** [comment]
I can try taking a look at some point if necessary. This seems like an issue with the transformer having different results to previous versions I would guess, though. 

Given TSFresh requires `statsmodels>=0.13` maybe the new ones are the correct values? I would normally say that the tests are passing doesn't mean it is working correctly, but in this case I would just trust TSFresh developers and update the expected values to the new results.

It's not the best currently, but there is a space setup in `contrib/_classification_test_reproduction.py` to help with updating the expected results.

**Comment by lmmentel** [comment]
Thanks for the offer to look into this. This is currently blocking #1620 which I was planning to get done within the next week, provided that I won't find any more surprises on the way. Let me know if you would be able to have look within that timeline otherwise I can fix this as mentioned above.

**Comment by fkiraly** [comment]
😭 let's maybe finish #1620 quickly and be cautious adding more estimators.
There's clearly some issues not being picked up by our current testing framework.

@lmmentel, I'd suggest, just escape all the failing tests for now, and let's investigate them one-by-one post-#1620.

@MatthewMiddlehurst, regarding the tests, I'm also generally critical of tests that ask methods to arrive at specific predictions up to the x-th decimal. These can be impacted by random number generators and idiosyncrasies of packages or python versions. As long as this is a general problem with the python ecosystem, we shouldn't be tying ourselves too strictly to consistency across versions imo.

**Comment by lmmentel** [comment]
I've updated the expected values for the tests after bumping the dependency so this can be closes after #1620. I'll add a statement to the PR description to close this automatically.
